kernel/api/src/main/java/org/sakaiproject/messaging/api/model/UserNotification.java

```diff
@@ -92,4 +92,7 @@ public class UserNotification implements PersistableEntity<Long> {
 
     @Transient
     private String tool;
+
+    @Transient
+    private Boolean isNotification = Boolean.TRUE;
```

Comment:
Can UserNotification be anything else than an notification? This seems quite confusing to me. Maybe the variable name could be different or a comment added?

Reply:
Yes, it can in fact. A UserNotification can actually just be a message directly to a tool's js. So, not intended to be a popup type thing or even go into the Notifications area. It now seems a bad name, I agree.

Reply:
Could you expand a bit on that use-case? It seems in this PR you don't make use of this feature. To me it's a bit odd, that we would update tool js when the user might not even have Sakai open.

Reply:
I've removed it. I added it to just make it clear in future that push is not just about visible notifications in the notifications area. The use case is this: somebody creates a new topic in conversations, the conversations service calls userMessagingService.message with a push message containing the new topic, or a reference to it, and that message is broadcast to all the relevant push subscriptions. The conversations js registers a push callback for those conversations messages and updates the client side ui in real time.

Reply:
We'd probably add a presence check for the tool messages. Think of tool messages as strictly online messages. Or, they could even be sent at all times and the service worker caches them. Anyway, some more thought is needed on that front, but live updating of tools is desirable.

Reply:
I'm working on another branch that starts the process of add progressive web app capabilities, based on this work.

Reply:
I'm a bit skeptical regarding this approach, but it might be that I just don't know enough about this the push api. If you have any resources regarding push messages that are not notifications and best practices around it, it would be great if you could share them with me, I didn't find much about it. What you describe sounds like a good use-case for websockets to mee, but they might be outdated?
You can resolve this comment though.

Reply:
websockets are not suitable for our use. They keep a connection open so are more resource intensive on our servers.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Push_API

The beauty of push is that we don't have to worry about how the message is delivered to the subscribed device. The vendor's push service deals with that - all we do is send the encrypted message to the push service and forget about it. Very scalable from our point of view.

Reply:
Thanks!